nomad/job_endpoint.go

```diff
@@ -433,6 +438,53 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 	return nil
 }
 
+// specChanged checks to see if the job spec has changed. If the job is multiregion,
+// it checks all regions. This handles the case of a multiregion job that has been
+// stopped in one or more regions being restarted via the run command.
+func (j *Job) specChanged(existingJob *structs.Job, args *structs.JobRegisterRequest) (bool, error) {
+	if existingJob == nil {
+		return false, nil
+	}
+
+	if existingJob.SpecChanged(args.Job) {
+		return true, nil
+	}
```

Comment:
This could be pulled out to the caller or the OSS stub function could consist of just this block, leaving it clear that there are two different things going on here: we're checking whether the spec changed _and_ we're checking other regions.